pkg/updater/resultstore/resultstore.go

```diff
@@ -72,19 +89,101 @@ func ResultStoreColumnReader(client *DownloadClient, reprocess time.Duration) up
 		})
-		for _, result := range results {
+
+		for _, pr := range processedResults {
 			// TODO: Make group ID something other than start time.
-			inflatedCol := processGroup(result)
+			inflatedCol := processGroup(pr)
 			receivers <- *inflatedCol
 		}
 		return nil
 	}
 }
 
+func processRawResults(log logrus.FieldLogger, results []*fetchResult) []*processedResult {
+	var processedResults []*processedResult
+	for _, result := range results {
+		pr := processRawResult(log, result)
+		processedResults = append(processedResults, pr)
+	}
+	return processedResults
+}
+
+// processRawResult converts raw fetchResults to processedResults with single action/target result/configured target result per targetID
+// Will skip processing any entries without Target or ConfiguredTarget
+func processRawResult(log logrus.FieldLogger, result *fetchResult) *processedResult {
+
+	multiActionResults := collateRawResults(log, result)
+	singleActionResults := isolateActions(log, multiActionResults)
+
+	return &processedResult{result.Invocation, singleActionResults}
+}
+
+// collateRawResults collates targets, configured targets and multiple actions into a single structure using targetID as a key
+func collateRawResults(log logrus.FieldLogger, result *fetchResult) map[string]*multiActionResult {
+	multiActionResults := make(map[string]*multiActionResult)
+	for _, target := range result.Targets {
+		trID := target.GetId().GetTargetId()
+		tr, ok := multiActionResults[trID]
+		if !ok {
+			tr = &multiActionResult{}
+			multiActionResults[trID] = tr
+		} else if tr.TargetProto != nil {
+			logrus.WithField("id", trID).Debug("Found duplicate target where not expected.")
+		}
+		tr.TargetProto = target
+	}
+	for _, configuredTarget := range result.ConfiguredTargets {
+		trID := configuredTarget.GetId().GetTargetId()
+		tr, ok := multiActionResults[trID]
+		if !ok {
+			tr = &multiActionResult{}
+			multiActionResults[trID] = tr
+			logrus.WithField("id", trID).Debug("Configured target doesn't have corresponding target?")
+		} else if tr.ConfiguredTargetProto != nil {
+			logrus.WithField("id", trID).Debug("Found duplicate configured target where not expected.")
+		}
+		tr.ConfiguredTargetProto = configuredTarget
+	}
+	for _, action := range result.Actions {
+		trID := action.GetId().GetTargetId()
+		tr, ok := multiActionResults[trID]
+		if !ok {
+			tr = &multiActionResult{}
+			multiActionResults[trID] = tr
+			logrus.WithField("id", trID).Debug("Action doesn't have corresponding target or configured target?")
+		}
+		tr.ActionProtos = append(tr.ActionProtos, action)
+	}
+	return multiActionResults
+}
+
+// isolateActions splits multiActionResults into one per action
+// Any entries without Target or ConfiguredTarget will be skipped
+func isolateActions(log logrus.FieldLogger, multiActionResults map[string]*multiActionResult) map[string][]*singleActionResult {
```

Comment:
nit: Might make this 'splitActions' (ala 'splitCells()' elsewhere in the Updater)?